src/api/v1/community_businesses/volunteer_logs.ts

```diff
@@ -99,6 +99,134 @@ const routes: Hapi.ServerRoute[] = [
+      auth: {
+        strategy: 'standard',
+        access: {
+          // TO BE REPLACED: "parent" with "own"
+          scope: ['volunteer_logs-parent:write'],
+        },
+      },
+      validate: {
+        payload: Joi.array().items(Joi.object({
+          activity: Joi.string().required(),
+          duration: Joi.object({
+            hours: Joi.number().integer().min(0),
+            minutes: Joi.number().integer().min(0),
+            seconds: Joi.number().integer().min(0),
+          }).required(),
+          startedAt: Joi.date().iso().max('now').default(() => new Date().toISOString(), 'now'),
+        })).required(),
+      },
+      response: { schema: response },
+      pre: [
+        { method: getCommunityBusiness, assign: 'communityBusiness' },
+      ],
+    },
+    handler: async (request: SyncMyVolunteerLogsRequest, h) => {
+      const {
+        server: { app: { knex } },
+        auth: { credentials: { user } },
+        pre: { communityBusiness },
+        payload,
+      } = request;
+
+      try {
+        await knex.transaction(async (trx) =>
+          Promise.all(payload.map(async (log) =>
+            VolunteerLogs.add(knex, {
+              ...log,
+              organisationId: communityBusiness.id,
+              userId: user.id,
+            })
+          ))
+        );
+
+        return null;
+
+      } catch (error) {
+        if (error.code === '23502') { // Violation of null constraint implies invalid activity
+          return Boom.badRequest('Invalid activity');
+        }
+        if (error.code === '23505') { // Violation of unique constraint => duplicate log
+          return Boom.badRequest('Duplicate logs in payload, check start times');
+        }
+        throw error;
+      }
+    },
+  },
+
+  {
+    method: 'GET',
+    path: '/community-businesses/me/volunteer-logs/summary',
+    options: {
+      description: 'Retreive summary of volunteer statistics from own organisation',
+      auth: {
+        strategy: 'standard',
+        access: {
+          // TO BE REPLACED: "volunteer_logs-parent" with "volunteer_logs-child"
+          scope: [
+            'volunteer_logs-parent:read',
+            'volunteer_logs-own:read',
+            'organisations_details-parent:read',
+            'organisations_details-own:read',
+          ],
+        },
+      },
+      response: { schema: response },
+      pre: [
+        { method: getCommunityBusiness, assign: 'communityBusiness' },
+      ],
+    },
+    handler: async (request, h) => {
+      const {
+        server: { app: { knex } },
+        auth: { credentials: { scope } },
+        pre: { communityBusiness },
+      } = request;
+
+      /*
+       * This manual checking of scopes is necessary because
+       * hapi.js scopes specification isn't capable of capturing
+       * complex scope rules e.g ((x AND y) OR (a AND b)).
+       *
+       * See: https://hapijs.com/api#-routeoptionsauthaccessscope
+       */
+      const hasParentScopes =
```

Comment:
Can this all be removed when I change the permissions?

Reply:
I don't think so, because these checks aren't about which permissions are associated to which roles, but about specifying more complex scope requirements than hapi can natively support.

When we change `parent` to `child`, we'll still need to check that they have both 'org_details' and 'volunteer_logs', unless we decide that the 'org_details' scope is unnecessary.

Reply:
Why are there so many scopes? I tried looking at the `api.json` but this route isn't in there 🙃

Reply:
lol, oops.

So my reasoning was that `org_details` was necessary here because we're querying "how many volunteers are there at the CB?", which doesn't seem like a `volunteer_logs` related question.

We _could_ overlook that in this instance, and then I'd only need `volunteer_logs` scopes and could get rid of the extra logic. 🙈

Reply:
would `volunteer_logs-sibling:read` cover this use case? To me that says I have permission to read the logs of other users at this org.

Reply:
But normal volunteers don't have that scope, right? From what I remember, there's no indication in the native app code that this feature is only for admins (but maybe we can just say that it will be?)

Reply:
I'm not following. Where on the volunteer app is this endpoint used?

Reply:
Only on the dashboard controller

Reply:
I'd be tempted to stick with `organisations_details-parent:read`. I think of logs as data related to a user, this info is annoymised and summarised making it have more in common with generic cb data, than the concept of volunteer_logs.

Reply:
OK, so just to be clear, we're saying we'll stick with the 4 scopes and the additional logic? I'm happy with that, I just want to know whether there's any remaining action here.

Reply:
No, I was suggesting dropping the `volunteer_logs` scopes.